rasp/base.py

```diff
@@ -16,23 +13,27 @@ def cleanup(self):
 
 
 class DefaultEngine(Engine):
-    def __init__(self, data=None, headers=None):
-        self.data = data
+    def __init__(self, headers=None):
+        self.session = self._session()
         self.headers = headers or {'User-Agent': DEFAULT_USER_AGENT}
```

Comment:
If these are really default headers, and not added to the headers passed directly to get_page_source (since you added headers there as well), then we should rename the instance variable to `default_headers`.   Otherwise, they should get merged in that method.